### PR TITLE
Unzip CWL Dependencies and make available to pre-processing

### PR DESCRIPTION
**Pull request by danbills**
I'm throwing in some nice-to-haves here:

* deleted unused "mungeId" artifact before FullyQualifiedName was born
* added "sbt-pack" SBT plugin which makes nice run scripts and doesn't assemble fat jar (slow)

### CODE REVIEW

File: engine/src/main/scala/cromwell/engine/workflow/lifecycle/MaterializeWorkflowDescriptorActor.scala [hunk 1]
```diff
@@ -434,22 +435,26 @@ class MaterializeWorkflowDescriptorActor(serviceRegistryActor: ActorRef,
...
-      for {
-        cwl <- CwlDecoder.decodeAllCwl(cwlFile)
-        wf <- fromEither[IO](cwl.select[Workflow].toRight(NonEmptyList.one(s"expected a workflow but got a $cwl")))
-        executable <-  fromEither[IO](wf.womExecutable(Option(source.inputsJson)))
-        graph <- fromEither[IO](executable.graph.toEither)
-        ioFunctions = new WdlFunctions(pathBuilders)
-        validatedWomNamespace <- fromEither[IO](validateWomNamespace(executable, ioFunctions))
-      } yield validatedWomNamespace
-    } finally {
-      cwlFile.delete(swallowIOExceptions = true)
```
Comment (danbills) [inline engine/src/main/scala/cromwell/engine/workflow/lifecycle/MaterializeWorkflowDescriptorActor.scala]: This `finally` block was for some reason executing before the parsing step, throwing `FileNotFound` error from `cwltool`.  

The file and the dependencies are being created in a temporary directory though, and they're small, so I don't think this is that bad.

I am thinking of another way to ensure they get deleted but short of using fs2 I don't know how to ensure resource safety without the (problematic) `finally` block.

**Comment by mcovarr** [comment]
Can you please open this in the main Cromwell PR?

**Comment by mcovarr** [comment]
Er, what I meant to write was, can you please create this branch in the main Cromwell repo and PR that instead? Having PRs in private repos has been causing review and maintenance pain.

**Comment by danbills** [comment]
Opening from main code repo

**Pull request closed without merging**